network/network-address/src/lib.rs

```diff
@@ -270,7 +270,7 @@ impl Arbitrary for RawNetworkAddress {
 ////////////////////
 
 impl NetworkAddress {
-    fn new(protocols: Vec<Protocol>) -> Self {
+    pub fn new(protocols: Vec<Protocol>) -> Self {
```

Comment:
ahh that's because it's trying to maintain the invariant that `protocols` is always non-empty. that's why `fn new` is private and only `TryFrom<Vec<Protocol>>` is exposed, which will return an `EmptyError` if the input is empty.

Reply:
damn, i thought i documented this... but it seems i missed it :sob:

Reply:
Interesting, we could enforce the invariant here on construction?